src/lib.rs

```diff
@@ -351,3 +435,71 @@ fn convert_response_to_query_response(res: skar_client::QueryResponse) -> Result
         },
     })
 }
+
+fn convert_response_to_pyarrow<'py>(
+    py: Python<'py>,
+    res: skar_client::QueryResponse,
+) -> Result<QueryResponseArrow> {
+
+    let blocks = res.data.blocks;
+    let transactions = res.data.transactions;
+    let logs = res.data.logs;
+
+    let (blocks, transactions, logs) = Python::with_gil(|py| {
```

Comment:
should be able to do it without `with_gil` since we already have the Python<'py> in the function arguments